### ISSUE TITLE
Proposing updates to the Toggle theme API

### ISSUE THREAD

**Issue by phinch** [grommet/grommet#5508]
Hi! This isn't a bug report but rather a forum to get some feedback on some expansions I'd like to make to the Toggle component's theming capability. I'm a new contributor but I'd be more than happy to take on these changes if they seem sound, as they would overlap well with stuff I'm working on at my company, which is a Grommet consumer. I read in the contribution guide that opening an issue is a good way to discuss plans, so here I am :)

### Problem

The [current Checkbox Toggle styles](https://github.com/grommet/grommet/blob/master/src/js/components/CheckBox/StyledCheckBox.js#L123) are coupled in a few ways (but not in all ways) to the checkbox style - e.g. the width of the toggle is determined by `theme.checkBox.toggle.size`, but the height is determined by `theme.checkBox.size`; the size of the knob and style of the border are also coupled to the checkBox theme in similar ways. We've found it difficult to implement a toggle to design specifications without extensive use of the `extend` properties, sometimes to overwrite styles being applied because of what we put in the theme to style our checkboxes.

### Proposed Changes

Given that the checkbox and toggle are visually very different components, I think it would be nice for Grommet to expose more toggle-specific theming options, and to use those options for styling (while falling back to checkbox styles where necessary to preserve backward compatibility). Specifically, I think I'd like to add the following to `theme.checkBox.toggle` (this all in addition to the existing theming API):
```
height?: string // height of the toggle background
// border for the toggle background
border?: {
  color?: ColorType;
  width?: string;
};
knob: {
  size?: string // size of the knob
}
```

I believe these additions would be sufficient to completely decouple the toggle styles from the checkbox styles. That said, as I mentioned above, we could coalesce back to the checkbox styles being used before, so this would not have to be a breaking change. My hope is that this would give a lot more clarity and ease for apps that use both a checkbox and a toggle while expecting them to look quite different.

Please let me know any thoughts that you have on this! Like I said, I haven't contributed before, so I hope I've gone about this in the right way and am open to feedback of any sort :) Looking forward to hearing from you all.

**Comment by ericsoderberghp** [comment]
I'm supportive of the general concept.

I want to think through the details of the contents of `theme.checkBox.toggle` a bit more. For instance, I could see `theme.checkBox.toggle.container` that would then have `background`, `border`, `round`, `size` keys and then `theme.checkBox.toggle.knob` that has similar keys.

**Comment by ericsoderberghp** [comment]
Good additions @halocline !

**Comment by phinch** [comment]
@ericsoderberghp thanks for these thoughts! I was definitely thinking pretty narrow for what would be needed for our use case, I like these suggestions. I was pretty focused on backward compatibility in my original thinking - is that something we should be sensitive to? I love the idea of having a `toggle.container` section for the container props and would be curious if we would want to also continue to support the top-level `toggle.background`, `.radius` etc. (of course it's possible either way, but wanted to ask if backward compatibility is a priority since these changes might make the theme type somewhat hard to follow)

**Comment by phinch** [comment]
@halocline thanks for the feedback :) here are some screenshots of each! The toggle height is 24px whereas the checkbox size is 20px; and the knob size is 16px. There's also some atypical stuff in the toggle with the active/inactive color: it's the background rather than the knob that changes color when active. I was also wondering about the `checked` state, which would definitely be used by us! it seems like the prevailing pattern in the Grommet checkbox is to use `extend: ({ checked }) => ...` to incorporate checked styles, which is definitely workable - but I think a `checked` section would be cool too if that sounds kosher.

our checkbox design:
![image](https://user-images.githubusercontent.com/9158038/128235943-82c5946c-4093-4cdc-aeb3-cc53928b7a50.png)
![image](https://user-images.githubusercontent.com/9158038/128235962-703e6cf5-74fa-4fcb-9182-af1f701ac818.png)
![image](https://user-images.githubusercontent.com/9158038/128236419-f6dfd577-015d-4433-8c7a-f10a5cb72f29.png)


toggle design:
![image](https://user-images.githubusercontent.com/9158038/128235890-ed2e8d93-edd9-4d89-8306-7a198aa37954.png)
![image](https://user-images.githubusercontent.com/9158038/128235910-30a8b50b-03ef-4405-8745-c0dc7b647ffe.png)
![image](https://user-images.githubusercontent.com/9158038/128236498-c3f81a3f-5b76-4f59-91f1-daea514808b3.png)

**Comment by halocline** [comment]
> here are some screenshots of each! 

Thank you, @phinch. Super helpful to have on hand.

**Comment by halocline** [comment]
> I was pretty focused on backward compatibility in my original thinking - is that something we should be sensitive to? I love the idea of having a `toggle.container` section for the container props and would be curious if we would want to also continue to support the top-level `toggle.background`, `.radius` etc. (of course it's possible either way, but wanted to ask if backward compatibility is a priority since these changes might make the theme type somewhat hard to follow)

Yes, backwards compatibility is definitely something to which we should be sensitive. Your thoughts here are spot on for the need to support the current top-level, plus concerns for clarity in following the theme type. I think there was a recent PR addressing a similar improvement to the theme but maintaining backwards compatibility, but am struggling to find it at the moment.

**Comment by phinch** [comment]
got it, sounds good :) thanks @halocline. i'll plan to start implementing a first pass at this tomorrow, taking all of the feedback into account! (will still be watching here for any other comments too)
